sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java

```diff
@@ -239,8 +246,53 @@ private TestPipeline(
   public Statement apply(final Statement statement, final Description description) {
     return new Statement() {
 
+      private final Predicate<Annotation> hasCategoryAnnotation = new Predicate<Annotation>() {
```

Comment:
That's a good point.

Since `hasCategoryAnnotation` is used only in the enclosing (anonymous) `Statement` instance I thought it would be nice to place it in the most inner scope, i.e. the anonymous `Statement` class, in which case it cannot be static due to it being anonymous.

Do you believe making it a constant outweighs the scoping consideration?